File: test_api.py
import paramiko
client = paramiko.SSHClient()
client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
client.connect('185.10.93.73', 22, 'root', 'vnmBXK1LnKBD!')

script = """
import urllib.request
import json
data = b'{"oddsHome": 2.50, "oddsDraw": 3.00, "oddsAway": 2.80, "league": "Premier League"}'
req = urllib.request.Request('http://localhost:8080/api/matches/find-similar', data=data, headers={'Content-Type': 'application/json'})
try:
    with urllib.request.urlopen(req) as resp:
        res = json.loads(resp.read())
        print(f"Returned {len(res)} matches")
except Exception as e:
    print("Error:", e)
"""

with client.open_sftp() as sftp:
    with sftp.file('/tmp/test_api.py', 'w') as f:
        f.write(script)

_, out, err = client.exec_command('python3 /tmp/test_api.py')
print("OUT:", out.read().decode('utf-8').strip())
client.close()

File: artifacts/api-server/test-api.mjs

const q1 = await fetch('http://localhost:8080/api/matches/find-similar', {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: JSON.stringify({
    oddsHome: 2.1, oddsDraw: 3.2, oddsAway: 3.1, league: 'GREEK CUP QUALIFICATION', oddsType: 'CLOSING'
  })
}).then(r => r.json());

const q2 = await fetch('http://localhost:8080/api/matches/find-similar', {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: JSON.stringify({
    oddsHome: 1.5, oddsDraw: 4.0, oddsAway: 5.0, league: 'PREMIER LEAGUE', oddsType: 'CLOSING'
  })
}).then(r => r.json());

console.log('Q1 top:', q1[0]?.match?.homeTeam, q1[0]?.match?.awayTeam, 'Score:', q1[0]?.similarityScore);
console.log('Q2 top:', q2[0]?.match?.homeTeam, q2[0]?.match?.awayTeam, 'Score:', q2[0]?.similarityScore);

